### DISCUSSION TITLE
Query events with filtering via API

### DISCUSSION THREAD

**Discussion by abudis** [datalust/seq-tickets#1654] [Help] [answered]
Hey all,

I'm trying to understand if it's possible to query for events, but with filtering applied via the API.

I see that there's GET /events endpoint available, but I don't see any option to filter for events.

Thanks!

**Comment by nblumhardt** [comment, marked as answer]
Hi Artem, thanks for getting in touch.

If you're using C# to connect to the API, the easiest way to do this is with the _Seq.Api_ client, which you can find at:

https://github.com/datalust/seq-api

The `SeqConnection.Events` object has all of the functions you'll need for this.

Otherwise; the `api/events/signal` endpoint is the place to start:

```
https://logs.datalust.co/api/events/signal?count=1&filter=@Message%20like%20%27test%27
```

The results will often come back as `Status: "Partial"`, in which case you can request the next page by passing the `&afterId=...` parameter with the value of `LastReadEventId` from the previous search result.

The C# client is definitely the way to go if you have the option.

Hope this helps!

Nick

**Reply by abudis** [reply]
Hey Nick,

Thanks for the speedy response!

Unfortunately I'm using Python as the client language, therefore I won't be able to the use the Seq.Api client.

I will definitely try out the latter option though, thanks a lot!

Best,
Artem

**Reply by nblumhardt** [reply]
Hi Artem,

In case it helps, Seq's UI uses the same API that client applications use, so checking out your browser's Network tab after issuing a few searches can highlight what API calls are made for different queries.

Nick

**Reply by abudis** [reply]
Thanks for the helpful tips Nick!

I actually managed to solve this yesterday with the help of your original post.

Artem
